vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -850,15 +900,36 @@ func resourceVSphereVirtualMachineCustomizeDiff(d *schema.ResourceDiff, meta int
 		return err
 	}
 
-	// Validate and normalize disk sub-resources
-	if err := virtualdevice.DiskDiffOperation(d, client); err != nil {
-		return err
+	// Validate and normalize disk sub-resources when not deploying from ovf
+	if len(d.Get("ovf_deploy").([]interface{})) == 0 {
+		if err := virtualdevice.DiskDiffOperation(d, client); err != nil {
+			return err
+		}
 	}
 	// When a VM is a member of a vApp container, it is no longer part of the VM
 	// tree, and therefore cannot have its VM folder set.
 	if _, ok := d.GetOk("folder"); ok && vappcontainer.IsVApp(client, d.Get("resource_pool_id").(string)) {
 		return fmt.Errorf("cannot set folder while VM is in a vApp container")
 	}
+
+	if len(d.Get("ovf_deploy").([]interface{})) > 0 {
```

Comment:
It would be good to add checks for the other ovf-required fields here such ash `host_system_id`, `datastore_id`, `datacenter_id`. It might be good to add also that `datacenter_id` is only used for ovf deployment and add a check that it isn't set when not deploying an ovf.